File: evalio-ui/src/services/manager/entities/templates.ts
export interface Template {
  id: string;
  professor_id: string;
  created_at: Date;
  questions: Question[];
  subject_name: string;
  period: string;
  number: number;
  group_id: string;
}

export interface Question {
  question: number;
  answer: string;
}

export interface CreateTemplate {
  subject_name: string;
  period: string;
  number: number;
  group_id: string;
  questions: Question[];
}

File: backend/grader_analyzer/grader_analyzer/domain/grader/entities/templates.py
from datetime import datetime

import pytz
from pydantic import BaseModel


class Question(BaseModel):
    question: int
    answer: str


class TemplateResponses(BaseModel):
    professor_id: str
    created_at: datetime = datetime.now(pytz.timezone("America/Bogota"))
    questions: list[Question]
    subject_name: str
    period: str
    number: int
